velox/vector/VariantToVector.cpp

```diff
@@ -89,6 +111,12 @@ ArrayVectorPtr variantArrayToVector(
   }
 
   const auto elementType = variantArray.at(nonNullIndex.value()).inferType();
+  if (elementType->kind() == TypeKind::SHORT_DECIMAL) {
+    return variantArrayToVectorImpl<TypeKind::SHORT_DECIMAL>(
+        variantArray, pool);
+  } else if (elementType->kind() == TypeKind::LONG_DECIMAL) {
+    return variantArrayToVectorImpl<TypeKind::LONG_DECIMAL>(variantArray, pool);
+  }
   return VELOX_DYNAMIC_SCALAR_TYPE_DISPATCH(
       variantArrayToVectorImpl, elementType->kind(), variantArray, pool);
```

Comment:
add a second argument to `variantArrayToVectorImpl`: `const TypePtr& type` and pass elementType

Reply:
Updated, thanks!